com.unity.visualeffectgraph/Editor/Models/VFXGraph.cs

```diff
@@ -21,19 +21,42 @@ namespace UnityEditor.VFX
     [InitializeOnLoad]
     class VFXGraphPreprocessor : AssetPostprocessor
     {
-        void OnPreprocessAsset()
+        static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
         {
-            bool isVFX = VisualEffectAssetModicationProcessor.HasVFXExtension(assetPath);
-            if (isVFX)
+            List<string> assetToReimport = null;
+
+            foreach (var assetPath in importedAssets)
             {
-                VisualEffectResource resource = VisualEffectResource.GetResourceAtPath(assetPath);
-                if (resource == null)
-                    return;
-                VFXGraph graph = resource.graph as VFXGraph;
-                if (graph != null)
-                    graph.SanitizeForImport();
-                else
-                    Debug.LogError("VisualEffectGraphResource without graph");
+                bool isVFX = VisualEffectAssetModicationProcessor.HasVFXExtension(assetPath);
+                if (isVFX)
+                {
+                    VisualEffectResource resource = VisualEffectResource.GetResourceAtPath(assetPath);
+                    if (resource == null)
+                        return;
+                    VFXGraph graph = resource.graph as VFXGraph;
+                    if (graph != null)
+                    {
+                        bool wasGraphSanitized = graph.sanitized;
+                        graph.SanitizeForImport();
+                        if (!wasGraphSanitized && graph.sanitized)
```

Comment:
Maybe we also want to check whether the graph was dirtied during the sanitization process? If it wasnt it means we don't need to reimport right?

Reply:
Actually, even if nothing has been dirtied during sanitization, we still need to reimport because we previously early skipped the compilation (awaiting to be allowed to sanitize the vfx). See [here](https://github.com/Unity-Technologies/Graphics/pull/4971/files#diff-ea3502fe33557cb681c8e37fabdbf3870b6f1ca4e60221ff653ecc699cdcf885R87)

Reply:
Ah yes sure ! That double import is really not ideal...